File: 78-subsets/78-subsets.go
func subsets(nums []int) [][]int {
  n := len(nums)
  ans := [][]int{}
  var backtrack func(path int, i int)
  backtrack = func(path int, i int) {
    res := []int{}
    for j, p := 0, path; p > 0; j, p = j+1, p>>1 {
      if (p & 1) > 0 { res = append(res, nums[j]) }
    }
    ans = append(ans, res)
    for j := i; j < n; j++ {
      backtrack(path | (1 << j), j+1)
    }
  }
  backtrack(0, 0)
  return ans
}

File: 78-subsets/78-subsets.js
/**
 * @param {number[]} nums
 * @return {number[][]}
 */
var subsets = function(nums) {
  const result = []
  const dfs = (path, i) => {
    const res = []
    for (let j = 0, p = path; p > 0; j++, p = p >> 1) {
      if (p & 1) res.push(nums[j])
    }
    result.push(res)
    for (let j = i; j < nums.length; j++) {
      dfs(path | (1 << j), j + 1)
    }
  }
  dfs(0, 0)
  return result  
};